cleanlab_pinecone_rag_workflow/Cleanlab_Pinecone_RAG.ipynb

```diff
@@ -0,0 +1,2673 @@
+   "source": [
+    "Now let's quey TLM using the prompt we previously created using our user query + document context passed from our RAG Pinecone index."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 196,
+   "id": "7c3f8830",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'response': 'In SAP Business One, item information is typically stored in the item master data. This master data includes details about each item, such as its description, pricing, inventory levels, and other relevant attributes. This information is then utilized throughout the sales process, from creating sales orders to invoicing and delivery, ensuring that all relevant data is consistently available and updated across the various documents involved in the sales workflow.',\n",
+       " 'trustworthiness_score': 0.9196129070099817}"
+      ]
+     },
+     "execution_count": 196,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "output = tlm.prompt(prompt)\n",
+    "output"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e9a8dcac-518d-420c-aee5-51f8566bfe31",
+   "metadata": {},
+   "source": [
+    "### Is this response correct?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 192,
+   "id": "6ece66fa-b3f5-4167-b9c4-7af3c265f336",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'Like customer master data, item information is stored in master data records.'"
+      ]
+     },
+     "execution_count": 192,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "df.iloc[78][\"text\"][3063:3140]\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7ee253b8-1ecf-4ea2-a423-13bbe016e11e",
+   "metadata": {},
+   "source": [
+    "### Can we find a hallucination?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 226,
+   "id": "214fe620",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'response': 'Yes, the TurboCharge EV6000 supports the CCS/CE/ISO charger protocols.',\n",
+       " 'trustworthiness_score': 0.6108415301123387}"
+      ]
+     },
+     "execution_count": 226,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "question = \"Does the TurboCharge EV6000 support the CCS/CE/ISO charger protocols?. Answer succinctly.\"\n",
+    "top_docs = \" | \".join(rag_pipeline.search(question, top_k=1, filter_query={\"topic\": {\"$eq\": \"product\"}})[0][0])\n",
+    "\n",
+    "\n",
+    "prompt = f\"Based on the following documents, answer the given question. \\n\\\n",
+    "Documents: {top_docs} \\n\\\n",
+    "Question: {question}\"\n",
+    "\n",
+    "# Runs the Cleanlab TLM with confidence reliablity scores on cheaper model with default quality preset\n",
+    "tlm_35_turbo = studio.TLM(options={\"model\":\"gpt-3.5-turbo-16k\"})\n",
```

Comment:
Why do we pick TLM 35? I would love if we could find a way not to have to intentionally use a bad LLM. GPT 4o mini is half the cost of GPT 3.5 turbo

if the problem is finding a hallucination, I could try to help get one...